garak/translator.py

```diff
@@ -0,0 +1,583 @@
+    
+    text = text.lower()
+    word_list = text.split()
+    if len(word_list) == 0:
+        return False
+
+    if len(word_list) >= 1:
+        word_list = remove_punctuation(word_list)
+    else:
+        word_list = word_list[0]
+
+    if word_list:
+        word_list = word_list.split()
+        cleaned_words = ' '.join(char for char in word_list if char.isalpha())
+        # Filter out empty strings
+        cleaned_words = cleaned_words.split()
+        cleaned_words = [word for word in cleaned_words if word]
+
+        if not cleaned_words:
+            return False
+
+        english_word_count = sum(1 for word in cleaned_words if word in english_words)
+        return (english_word_count / len(cleaned_words)) > 0.5
+    return False
+
+
+def split_input_text(input_text: str) -> list:
+    """Split input text based on the presence of ': '."""
+    if ": " in input_text:
+        split_text = input_text.splitlines()
+        split_text = [line.split(":") for line in split_text]
+        split_text = [item for sublist in split_text for item in sublist]
+    else:
+        split_text = input_text.splitlines()
+    return split_text
+
+
+def contains_invisible_unicode(text: str) -> bool:
+    """Determine whether the text contains invisible Unicode characters."""
+    if not text:
+        return False
+    for char in text:
+        if unicodedata.category(char) not in {'Cf', 'Cn', 'Zs'}:
+            return False
+    return True
+
+
+def is_nested_list(lst: list) -> bool:
+    """Check if the given list is a nested list."""
+    return any(isinstance(i, list) for i in lst)
+
+
+class SimpleTranslator:
+    """DeepL or NIM translation option"""
+
+    # Reference: https://developers.deepl.com/docs/resources/supported-languages
+    bcp47_deepl = [
+        "ar", "bg", "cs", "da", "de",
+        "en", "el", "es", "et", "fi",
+        "fr", "hu", "id", "it", "ja",
+        "ko", "lt", "lv", "nb", "nl",
+        "pl", "pt", "ro", "ru", "sk",
+        "sl", "sv", "tr", "uk", "zh"
+    ]
+    
+    # Reference: https://docs.nvidia.com/nim/riva/nmt/latest/support-matrix.html#models
+    bcp47_riva = [
+        "zh", "ru", "de", "es", "fr",
+        "da", "el", "fi", "hu", "it",
+        "lt", "lv", "nl", "no", "pl",
+        "pt", "ro", "sk", "sv", "ja",
+        "hi", "ko", "et", "sl", "bg",
+        "uk", "hr", "ar", "vi", "tr",
+        "id", "cs"
+    ]
+    
+    DEEPL_ENV_VAR = "DEEPL_API_KEY"
+    NIM_ENV_VAR = "NIM_API_KEY"
+
+    def __init__(self, plugin_generators_dict: dict={}, source_lang: str="en") -> None:
+        self.translator = None
+        self.nmt_client = None
+        self.translation_service = ""
+        self.source_lang = source_lang
+        self.target_lang = plugin_generators_dict.get("lang_spec", "en")
+        self.translation_service = plugin_generators_dict.get("translation_service", "")
+        self.deepl_api_key = os.getenv(self.DEEPL_ENV_VAR)
+        self.nim_api_key = os.getenv(self.NIM_ENV_VAR)
```

Comment:
these should come from config passed to the `SimpleTranslator` - see pattern in base generator, and the `Configurable` mixin

Reply:
OK, I'll fix it